packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/drp/dedicatedCloud-datacenter-drp.routing.js

```diff
@@ -23,16 +23,91 @@ export default /* @ngInject */ ($stateProvider) => {
         ),
       disableForUS: /* @ngInject */ ($q, isDrpAvailable) =>
         !isDrpAvailable ? $q.reject() : $q.when(),
+      getIpOrderLink: /* @ngInject */ ($state) => (
+        drpType,
+        isLegacyOrder,
+        pccStep,
+      ) =>
+        $state.href(
+          `app.dedicatedClouds.datacenter.drp.${drpType}.${pccStep}.${
+            isLegacyOrder ? 'legacyOrderIp' : 'orderIp'
+          }`,
+        ),
+      setDisableSuccessAlertPreference: /* @ngInject */ (dedicatedCloudDrp) => (
+        pccId,
+        value,
+      ) => dedicatedCloudDrp.setDisableSuccessAlertPreference(pccId, value),
       selectedDrpType: /* @ngInject */ ($transition$) => ({
         id: get($transition$.params().selectedDrpType, 'id', null),
       }),
+      setupConfiguration: /* @ngInject */ (
+        $q,
+        $translate,
+        $timeout,
+        $window,
+        dedicatedCloudDrp,
+        displayErrorMessage,
+        displayInfoMessage,
+        displaySuccessMessage,
+        goToPccDashboard,
+        setDisableSuccessAlertPreference,
+        storeZertoOptionOrderInUserPref,
+        DEDICATED_CLOUD_CONSTANTS,
+      ) => (drpInformations) =>
+        dedicatedCloudDrp
+          .enableDrp(
+            drpInformations,
+            drpInformations.primaryPcc.generation !==
+              DEDICATED_CLOUD_CONSTANTS.pccNewGeneration,
+          )
+          .then((enableDrp) => {
+            const storeOptionPromise = enableDrp.url
+              ? storeZertoOptionOrderInUserPref(drpInformations, enableDrp)
+              : $q.when();
+
+            return $q
+              .all([
+                storeOptionPromise,
+                setDisableSuccessAlertPreference(
+                  drpInformations.primaryPcc.serviceName,
+                  false,
+                ),
+              ])
+              .then(() => goToPccDashboard(true))
+              .then(() => {
+                if (!enableDrp.hasAutoPay) {
+                  displaySuccessMessage(
+                    `${$translate.instant(
+                      'dedicatedCloud_datacenter_drp_confirm_order',
+                      { billUrl: enableDrp.url },
+                    )}`,
+                  );
+                  $window.open(enableDrp.url, '_blank');
+                } else {
+                  displayInfoMessage(`
+                    ${$translate.instant(
+                      'dedicatedCloud_datacenter_drp_confirm_creation_pending',
+                    )} ${$translate.instant(
+                    'dedicatedCloud_datacenter_drp_confirm_creation_pending_task',
+                  )}
+                `);
+                }
+              });
+          })
+          .catch((error) => {
+            displayErrorMessage(
+              `${$translate.instant(
+                'dedicatedCloud_datacenter_drp_confirm_create_error',
+              )} ${get(error, 'data.message', error.message)}`,
+            );
+          }),
```

Comment:
This functions looks a bit big. Can it be chunked into smaller parts ? (I understand that some logic is linked to display so not everything can go to a service

Reply:
Yes, you're right it is big. What could be splitted, is the logic related to user preferences. But as I can't set the whole function into a service (with display logic as you raised the point), I choose to let it there